kroto-plus-coroutines/src/test/kotlin/com/github/marcoferrer/krotoplus/coroutines/integration/BidiStreamingTests.kt

```diff
@@ -0,0 +1,136 @@
+ */
+
+package com.github.marcoferrer.krotoplus.coroutines.integration
+
+import com.github.marcoferrer.krotoplus.coroutines.utils.assertFailsWithStatus
+import com.github.marcoferrer.krotoplus.coroutines.withCoroutineContext
+import io.grpc.Status
+import io.grpc.examples.helloworld.GreeterCoroutineGrpc
+import io.grpc.examples.helloworld.HelloReply
+import io.grpc.examples.helloworld.HelloRequest
+import io.grpc.examples.helloworld.send
+import io.grpc.testing.GrpcServerRule
+import io.mockk.coVerify
+import io.mockk.spyk
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.channels.toList
+import kotlinx.coroutines.debug.DebugProbes
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.junit.Rule
+import org.junit.Test
+import kotlin.coroutines.CoroutineContext
+import kotlin.test.assertEquals
+
+class BidiStreamingTests {
+
+    @[Rule JvmField]
+    var grpcServerRule = GrpcServerRule().directExecutor()
+
+    @[Rule JvmField]
+    var nonDirectGrpcServerRule = GrpcServerRule()
+
+    @Test
+    fun `Bidi streaming rendezvous impl completes successfully`() {
+        nonDirectGrpcServerRule.serviceRegistry.addService(object : GreeterCoroutineGrpc.GreeterImplBase(){
+
+            override val initialContext: CoroutineContext
+                get() = Dispatchers.Default
+
+            override suspend fun sayHelloStreaming(
+                requestChannel: ReceiveChannel<HelloRequest>,
+                responseChannel: SendChannel<HelloReply>
+            ) {
+                requestChannel.consumeEach { request ->
+                    repeat(3) {
+                        responseChannel.send { message = request.name }
+                    }
+                }
+                responseChannel.close()
+            }
+        })
+        runBlocking {
+
+            val stub = GreeterCoroutineGrpc.newStub(nonDirectGrpcServerRule.channel)
+                .withCoroutineContext()
+
+            val (requestChannel, responseChannel) = stub.sayHelloStreaming()
+
+            launch(Dispatchers.Default) {
+                repeat(3) {
+                    requestChannel.send { name = "name $it" }
+                }
+                requestChannel.close()
+            }
+
+            val results = responseChannel.toList()
+            assertEquals(9, results.size)
+
+            val expected = "name 0|name 0|name 0" +
+                    "|name 1|name 1|name 1" +
+                    "|name 2|name 2|name 2"
+            assertEquals(
+                expected,
+                results.joinToString(separator = "|") { it.message }
+            )
+        }
+    }
+
+    @Test
+    fun `Client cancellation cancels server rpc scope`() {
```

Comment:
@chris-blacker I think I was able to reproduce your hanging client issue with this test. It turns out that the outbound message handler wasnt bound to normal channel cancellation. Thats been fixed in the latest changes. Since the actor wasnt bound, client coroutines would hang since one of their children would never complete.